pkg/scheduler/plugins/gang/gang.go

```diff
@@ -223,6 +223,8 @@ func (gp *gangPlugin) OnSessionClose(ssn *framework.Session) {
 				klog.Errorf("Failed to update job <%s/%s> condition: %v",
 					job.Namespace, job.Name, err)
 			}
+
+			metrics.RegisterJobSuccess(job.Name)
```

Comment:
I am a little curious, why do we add indicators here? According to my understanding, should the indicators be added after the job runs successfully?

Reply:
https://github.com/volcano-sh/volcano/issues/2493#issue-1371690108
Maybe I have misunderstood?

Reply:
IMO. The metrics of success or failure should be determined when the controller state changes. The scheduler is only used to schedule to a certain node and does not care whether the pod runs successfully.

Reply:
Yeah maybe we should set them in job controller.